pkg/client/apiutil/apimachinery.go

```diff
@@ -61,9 +62,13 @@ func AddToProtobufScheme(addToScheme func(*runtime.Scheme) error) error {
 
 // NewDiscoveryRESTMapper constructs a new RESTMapper based on discovery
 // information fetched by a new client with the given config.
-func NewDiscoveryRESTMapper(c *rest.Config) (meta.RESTMapper, error) {
+func NewDiscoveryRESTMapper(c *rest.Config, httpClient *http.Client) (meta.RESTMapper, error) {
+	if httpClient == nil {
```

Comment:
the apiutil package is not actually internal but exported, can we return an error if this is nil here and in `RESTClientForGVK`?

I just find it somewhat debatable that we make this a mandatory arg now, since this can not be just any http client but must be constructed from the rest config.

Reply:
I made the parameter mandatory because we wanted to not default in multiple places (see comments above).
If we expect users to directly use this function, I would argue that defaulting wouldn't be a bad idea?
I'll create a PR to change the panic to an error & let's continue the discussion there.

Reply:
@inteon Looks like this one still resturns a panic. Was that intended?

Reply:
Sorry not this one. `NewDynamicRESTMapper` in dynamicrestmapper.go

Reply:
https://github.com/kubernetes-sigs/controller-runtime/pull/2164

Reply:
@sbueringer this PR fixes the issue for `NewDynamicRESTMapper` too: https://github.com/kubernetes-sigs/controller-runtime/pull/2178

Reply:
Thx!